cs/libdpr/samples/DprCounters/DprCounters/python/start_cluster.py

```diff
@@ -0,0 +1,250 @@
+            self.storage_capacity = storage_capacity
+            self.cpu_request = cpu_request
+            self.cpu_limit = cpu_limit
+            self.memory_request = memory_request
+            self.memory_limit = memory_limit
+
+    def __init__(self, checkpoint_dir:str = None, azure:bool = False) -> void:
+        config.load_kube_config()
+        self.core = client.CoreV1Api()
+        self.apps = client.AppsV1Api()
+        self.servers = []
+        self.azure = azure
+        if checkpoint_dir:
+            self.directory = checkpoint_dir
+        else:
+            if azure:
+                self.directory = self.AZURE_DIRECTORY
+            else:
+                self.directory = self.LOCAL_DIRECTORY
+
+    def addServer(self, type:str, storage_location:str = None, storage_capacity:str = None, cpu_request:str = None, cpu_limit:str = None, memory_request:str = None, memory_limit:str = None):
+        if type not in self.SUPPORTED_SERVER_TYPES:
+            raise ArgumentError("Unsupported server type") 
+        self.servers.append(self.Server(type, storage_location, storage_capacity, cpu_request, cpu_limit, memory_request, memory_limit))
+
+    def isDprFinderRunning(self) -> bool:
+        all_pods = self.core.list_namespaced_pod(namespace="default")
+        for pod in all_pods.items:
+            if pod.metadata.name == "dpr-finder-0":
+                return pod.status.phase == "Running"
+
+    def startDprFinder(self, dprPath:str = None) -> void:
+
+        def specifyImage(dprYaml):
+            containersList = dprYaml["spec"]["template"]["spec"]["containers"]
+            if self.azure:
+                containersList[0]["image"] = self.AZURE_IMAGE
+                containersList[0]["imagePullPolicy"] = self.AZURE_POLICY
+            else:
+                containersList[0]["image"] = self.LOCAL_IMAGE
+                containersList[0]["imagePullPolicy"] = self.LOCAL_POLICY
+
+
+        if not dprPath:
+            dprPath = os.path.join(self.directory, "yaml/DprFinder.yaml")
+        with open(dprPath) as f:
+            dprYaml = yaml.safe_load_all(f.read())
+        k = 0
+        for yaml_doc in dprYaml:
+            if k==0:
+                self.core.create_namespaced_service("default", yaml_doc)
+            elif k==1:
+                specifyImage(yaml_doc)
+                self.apps.create_namespaced_stateful_set("default", yaml_doc)
+            else:
+                raise ArgumentError("The yaml file provided is wrongly formatted")
+            k+=1
+
+    def killDprFinder(self) -> void:
+        self.core.delete_namespaced_pod(name="dpr-finder-0", namespace="default")
+
+    def stopDprFinder(self) -> void:
+        self.apps.delete_namespaced_stateful_set(name="dpr-finder", namespace = "default")
+        self.core.delete_namespaced_service(name="dpr-finder-svc", namespace="default")
+    
+    def startCounterServers(self, counterServerService:str = None, counterServiceStateful:str = None) -> void:
```

Comment:
Eventually we want to move away from only CounterServers and make it arbitrary state objects